src/port.js

```diff
@@ -114,6 +141,23 @@ export default class Port {
 		}
 
 	}
+	registerService(serviceType, version, service) {
+		if(this.isConnected) {
+			throw new Error('Register services before connecting');
+		}
+		if(!typeNameValidator.test(serviceType)) {
+			throw new Error(`Invalid service type "${serviceType}"`);
+		}
+		let methodNames = [];
+		for(let p in service) {
+			if(typeof(service[p]) === 'function') {
+				methodNames.push(p);
+				this.onRequest(`service:${serviceType}:${version}:${p}`, service[p]);
```

Comment:
@cpacey: was able to refactor the `registerService` call a fair bit based on your suggestions, thanks!